### PR TITLE
Fixes future prediction and other bugs

### PR DESCRIPTION
**Pull request by petersen-f**
This PR mainly fixes the prediction and adds options to make sure user can always predict the future as well as minor bugs:
- adds warning/error when covariates/factors but no training indicator is provided
- adds option for periodical prediction (same like prophet module) where future data frame for prediction is created that extends beyond provided data + warning that frequency and units should be the same as training data. this way user can predict future without having to provide training indicator/include in training variable
- forecast verification plot now also works with only one prediction model
- available models that can be computed are now based on qml logic and not R function which reduces bugs -> e.g. when no covariates are selected, user can only select autoregressive models
- fixes some minor things dependency issues and qml options as well as enabling weekly seasonality for the prophet module

### CODE REVIEW

File: inst/qml/predictiveAnalytics.qml [hunk 1]
```diff
@@ -371,51 +375,68 @@ Form
...
+						{
+							models[2] = {label : qsTr("linear regression - regression + lag"), value: "lmSpikeRegLag"}
+							models[10] = {label : qsTr("bart - regression + lag"), value: "bartRegLag"}
+						}
+						// function returns one empty value and i haven't figured out why, so this is workaround
+						const modelsFiltered = models.filter((obj) => obj.label !== '')
 
+						const modelsList = [{values: modelsFiltered}]
+						return(modelsList)
+					}
```
Comment (petersen-f) [inline inst/qml/predictiveAnalytics.qml:426]: This function filters the available prediction models depending on whether they are possible based on the users input. For example one shouldn't be able to add a regression model like BART when no covariates are provided.
However, I discovered that the (selected) models in the qml elements are not saved when the JASP file is reopened. So one doesn't know anymore what models were selected for verification and prediction (although the JASP output/results persist).

I will try to fix this, please wait with approving this PR until then.

Comment (petersen-f) [reply inst/qml/predictiveAnalytics.qml]: I fixed it by creating a invisible availableVariableList that uses the function to make the right models available and then just use this as a source for a second availableVariableList that the user can see and use. this is a bit hacky, but works..

**Comment by petersen-f** [comment]
This PR now closes issue [#2336](https://github.com/jasp-stats/INTERNAL-jasp/issues/2390),[#2390](https://github.com/jasp-stats/INTERNAL-jasp/issues/2339),[#2455](https://github.com/jasp-stats/jasp-test-release/issues/2455),[#2456](https://github.com/jasp-stats/jasp-test-release/issues/2456),[#2457](https://github.com/jasp-stats/jasp-test-release/issues/2457),[#2458](https://github.com/jasp-stats/jasp-test-release/issues/2458).

It adds quite a few warnings when inputs do not make sense, limits some of the qml options to sensible values and fixes some bugs that prevented computation.

File: R/predictiveAnalytics.R [hunk 2]
```diff
@@ -73,6 +75,48 @@ predictiveAnalytics <- function(jaspResults, dataset, options) {
...
+
+    },
+    .predictionArgsChecks <- function(){
+      if(length(options$covariates) + length(options$factors) > 0 && options$trainingIndicator == "" &&  options$futurePredPredictionType != "noFuturePrediction")
+        return(gettext(paste("When 'Covariates' or 'Factors' are provided, they also need to be supplied for the future prediction period.",
+        "Please provide the 'Include in Training' variable where a value of '1' indicates that this period is used for training/verification - and a value of '0' that it is used for prediction.",
+        "The values for the dependent variable are allowed to be missing for the prediction period.",
+        "Alternatively, you could remove the covariates and select the specific time period you want to predict under 'Future Prediction' -> 'Periodical'.",
+        "That way the data is automatically extended into the future based on your settings. \n", 
+        "If you just want to check how well the predictions perform historically you can choose the option 'No forecast - verification only'")))
```
Comment (vandenman) [inline R/predictiveAnalytics.R:96]: you cannot use `paste` inside `gettext`, please format this as a single string. The same goes for the check below.

File: R/predictiveAnalytics.R [hunk 3]
```diff
@@ -1196,9 +1233,9 @@ lagit <- function(a,k) {
                        )
 
   predSummary <- aperm(apply(X = predArray,c(1,3) ,
-                             function(x) c(mean = mean(x),
-                                           upr = quantile(x,0.975),
-                                           lwr = quantile(x,0.025)),
+                             function(x) c(mean = mean(x,na.rm = T),
+                                           upr = quantile(x,0.975, na.rm = T),
+                                           lwr = quantile(x,0.025, na.rm = T)),
```
Comment (vandenman) [inline R/predictiveAnalytics.R:1238]:
```suggestion
                             function(x) c(mean = mean(x,na.rm = TRUE),
                                           upr = quantile(x,0.975, na.rm = TRUE),
                                           lwr = quantile(x,0.025, na.rm = TRUE)),
```

File: R/predictiveAnalytics.R [hunk 4]
```diff
@@ -1476,28 +1518,30 @@ lagit <- function(a,k) {
 
     #BMA
 
-    if(!is.null(jaspResults[["predanResults"]][["bmaResState"]]) && "BMA" %in% options$"modelsToPlot"){
+    if( "BMA" %in% options$"modelsToPlot" && options$checkPerformBma){
+      try({
+        print("doing the bma plotting")
```
Comment (vandenman) [inline R/predictiveAnalytics.R:1523]:
```suggestion
```

File: R/predictiveAnalytics.R [hunk 5]
```diff
@@ -1779,29 +1839,38 @@ lagit <- function(a,k) {
...
+    futureFrame$tt <- (nrRows+1):(nrRows+nrow(futureFrame))
+
+    if(all(is.na(dataEng$y)) || var(dataEng$y) == 0){
       errorPlot <- createJaspPlot()
-      errorPlot$setError(gettext("Cannot compute forecast. Larger forecast horizon requested than indicated by 'Include in Training' variable. Reduce forecast horizon or change training indicator"))
+      errorPlot$setError(gettext(paste(
+        "Cannot train models for future prediction.",
+        "The data used for training contains only missing values or has a variance of zero, making prediction impossible",
+        "Either provide better data or change the training window for future prediction."
+        )))
```
Comment (vandenman) [inline R/predictiveAnalytics.R:1860]: no `paste` within `gettext`

**Review by vandenman — CHANGES_REQUESTED** [review]

**Comment by petersen-f** [comment]
@vandenman Thank you Don! I indeed found more examples where I did this wrong... But I fixed every gettext now.

**Review by vandenman — APPROVED** [review]

**Pull request merged**